STModCellular.cpp

```diff
@@ -73,37 +124,50 @@ nsapi_error_t STModCellular::soft_power_on() {
         _at->write_int(2);
         _at->write_int(2);
         _at->cmd_stop_read_resp();
-
         err = _at->get_last_error();
+        _at->unlock();
+
         if (err == NSAPI_ERROR_OK) {
             tr_debug("Flow control turned ON");
         } else {
             tr_error("Failed to enable hw flow control");
         }
-
-        _at->restore_at_timeout();
-        _at->unlock();
     }
 #endif
+
+    wait_ms(500);
+
+    _at->lock();
+    /*  Verify Flow Control settings */
+    _at->cmd_start("AT+IFC?");
+    _at->cmd_stop_read_resp();
+    _at->unlock();
+
     return err;
 }
+
 nsapi_error_t STModCellular::soft_power_off() {
     _at->cmd_start("AT+QPOWD");
     _at->cmd_stop();
     wait_ms(1000);
     // should wait for POWERED DOWN with a time out up to 65 second according to the manual.
     // we cannot afford such a long wait though.
-    return QUECTEL_UG96::soft_power_off();
+    return STMOD_CELLULAR_MODEM::soft_power_off();
 }
 
 #if MBED_CONF_STMOD_CELLULAR_PROVIDE_DEFAULT
 #include "UARTSerial.h"
 CellularDevice *CellularDevice::get_default_instance()
 {
+    tr_debug("MODEM default instance\r\n");
+
     static UARTSerial serial(MBED_CONF_STMOD_CELLULAR_TX, MBED_CONF_STMOD_CELLULAR_RX, MBED_CONF_STMOD_CELLULAR_BAUDRATE);
+#if defined (MBED_CONF_STMOD_CELLULAR_RTS) && defined(MBED_CONF_STMOD_CELLULAR_CTS)
```

Comment:
This shall actually be done after the `AT+IFC` command is issued. Otherwise we may endup in a locked state where the mcu waits for the modem to be ready to receive while it is actually a floating `CTS` pin.

Reply:
Hmm actually we need to initialize the UART before we start the UART transmissions ... 
this seems also to be done this way for all other modems in mbed-os tree:
https://github.com/ARMmbed/mbed-os/blob/master/features/cellular/framework/targets/QUECTEL/BG96/QUECTEL_BG96.cpp#L82

Reply:
If `AT+IFC` follows the same logic as `AT+IPR`, according to [Quectel's](https://www.quectel.com/UploadImage/Downlad/Quectel_BG96_AT_Commands_Manual_V2.1.pdf#%5B%7B%22num%22%3A107%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C51%2C621%2C0%5D) documentation, the configuration takes place after any result code is delivered.

All other modem's driver may suffer from the same issue ;-)

Reply:
> All other modem's driver may suffer from the same issue ;-)
I understand the point.  Still I haven't seen any issue with this sequence, and I guess developers of other modem drivers have checked the same. 

Anyway, I will git it a try and see if we can activate flow control after OK response.